src/GraphQl/Resolver/Stage/SerializeStage.php

```diff
@@ -175,14 +174,21 @@ private function serializeCursorBasedPaginatedCollection(iterable $collection, a
      */
     private function serializePageBasedPaginatedCollection(iterable $collection, array $normalizationContext): array
     {
-        if (!($collection instanceof PaginatorInterface)) {
-            throw new \LogicException(sprintf('Collection returned by the collection data provider must implement %s.', PaginatorInterface::class));
+        if (!($collection instanceof PartialPaginatorInterface)) {
+            throw new \LogicException(sprintf('Collection returned by the collection data provider must implement %s or %s.', PaginatorInterface::class, PartialPaginatorInterface::class));
         }
 
-        $data = $this->getDefaultPageBasedPaginatedData();
-        $data['paginationInfo']['totalCount'] = $collection->getTotalItems();
-        $data['paginationInfo']['lastPage'] = $collection->getLastPage();
-        $data['paginationInfo']['itemsPerPage'] = $collection->getItemsPerPage();
+        $data = [
+            'collection' => [],
+            'paginationInfo' => [
+                'itemsPerPage' => $collection->getItemsPerPage(),
+            ],
+        ];
+
+        if ($collection instanceof PaginatorInterface) {
```

Comment:
The serialized data change depending of the context, maybe we should check if it's the case for the GraphQL schema too: if pagination client partial is enabled, the `totalCount` and `lastPage` should be removed from the schema.